routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, "GetUserAssociations: problem getting UTXO view")
+		return
+	}
+
+	// Construct association query.
+	associationQuery, err := fes._constructUserAssociationQueryFromParams(utxoView, req.Body, AssociationQueryTypeQuery)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetUserAssociations: %v", err))
+		return
+	}
+
+	// Query for association entries.
+	associationEntries, err := utxoView.GetUserAssociationsByAttributes(associationQuery)
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("GetUserAssociations: %v", err))
+		return
+	}
+
+	// Convert AssociationEntries to AssociationResponses.
+	associationResponses := []*UserAssociationResponse{}
+	for _, associationEntry := range associationEntries {
+		associationResponses = append(associationResponses, &UserAssociationResponse{
+			AssociationID:                  hex.EncodeToString(associationEntry.AssociationID.ToBytes()),
+			TransactorPublicKeyBase58Check: lib.Base58CheckEncode(utxoView.GetPublicKeyForPKID(associationEntry.TransactorPKID), false, fes.Params),
+			TargetUserPublicKeyBase58Check: lib.Base58CheckEncode(utxoView.GetPublicKeyForPKID(associationEntry.TargetUserPKID), false, fes.Params),
+			AppPublicKeyBase58Check:        lib.Base58CheckEncode(utxoView.GetPublicKeyForPKID(associationEntry.AppPKID), false, fes.Params),
+			AssociationType:                string(associationEntry.AssociationType),
+			AssociationValue:               string(associationEntry.AssociationValue),
+			ExtraData:                      DecodeExtraDataMap(fes.Params, utxoView, associationEntry.ExtraData),
+			BlockHeight:                    associationEntry.BlockHeight,
+		})
+	}
+
+	// JSON encode response.
+	response := UserAssociationsResponse{Associations: associationResponses}
+	if err = json.NewEncoder(ww).Encode(response); err != nil {
+		_AddInternalServerError(ww, "GetUserAssociations: problem encoding response as JSON")
+		return
+	}
+}
+
+func (fes *APIServer) CountUserAssociations(ww http.ResponseWriter, req *http.Request) {
+	// Create UTXO view.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, "CountUserAssociations: problem getting UTXO view")
+		return
+	}
+
+	// Construct association query.
+	associationQuery, err := fes._constructUserAssociationQueryFromParams(utxoView, req.Body, AssociationQueryTypeCount)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("CountUserAssociations: %v", err))
+		return
+	}
+
+	// Count association entries.
+	count, err := utxoView.CountUserAssociationsByAttributes(associationQuery)
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("CountUserAssociations: %v", err))
+		return
+	}
+
+	// JSON encode response.
+	response := AssociationsCountResponse{Count: count}
+	if err = json.NewEncoder(ww).Encode(response); err != nil {
```

Comment:
we may just want to encode count as the response - it is probably just a touch faster and we don't really need a JSON object for this. additionally, we can axe the response struct definition.
```suggestion
	if err = json.NewEncoder(ww).Encode(count); err != nil {
```

Reply:
That feels a little wonky to me. IMO not saving that much on bandwidth. And I always default to returning a proper JSON object. If for nothing else, future-proofs it in case we add additional fields.